dataclasses_avroschema/fields.py

```diff
@@ -460,37 +463,52 @@ def fake(self) -> bytes:
 
 @dataclasses.dataclass
 class EnumField(BaseField):
+    _VALID_META_FIELDS = ["aliases", "default", "doc", "namespace"]
+
+    def _get_meta_class_attributes(self) -> typing.Dict[str, typing.Any]:
+        meta_dict = {}
+        if "Meta" in self.type.__members__:
+            meta = self.type.__members__["Meta"].value
+            meta_dict = {k: v for k, v in vars(meta).items() if k in self._VALID_META_FIELDS}
```

Comment:
This is a clever solution to collect metadata for a field. However, a `dataclass` approach would be simpler in my opinion.

Consider something similar to the mechanism of collecting metadata in the `AvroModel`:
```python
CT = TypeVar("CT", bound="FieldMetadata")


@dataclass
class FieldMetadata:
    aliases: Optional[list[str]] = None
    default: Optional[str] = None
    doc: Optional[str] = None
    namespace: Optional[str] = None

    @classmethod
    def create(cls: type[CT], klass: type) -> CT:
        return cls(
            aliases=getattr(klass, "aliases", None),
            default=getattr(klass, "default", None),
            doc=getattr(klass, "doc", None),
            namespace=getattr(klass, "namespace", None),
        )
```

However, it would probably be worth refactoring other fields to use this approach. It might be out of the scope of this PR. What do you guys think?

Reply:
Yes, I completely agree that the approach that you point is better, however I think we should  do it in another `PR`. We will need this at the moment of refactor the `Decimal` and `Fixed` types if I am not mistaken